src/utils/urls.js

```diff
@@ -16,34 +16,54 @@
  */
 
 /**
- * Get the application base URL (JupyterHub). Always includes trailing slash.
- * @returns {string} - application base URL (JupyterHub) with trailing slash
+ * Remove double forward-slashes from URL's. It avoids the slashes that
+ * are preceded by ':', so that the slashes in the URL protocol are kept.
+ *
+ * @private
+ * @param {string} url - the URL
+ * @returns {string} - a URL without unnecessary double forward-slashes
  */
-export function getBaseUrl () {
-  const port = window.location.port ? ':' + window.location.port : ''
-  const baseUrl = `${window.location.protocol}//${window.location.hostname}${port}${window.location.pathname}`
-  if (baseUrl.endsWith('//')) {
-    return baseUrl.substring(0, baseUrl.length - 1)
-  } else if (baseUrl.endsWith('/')) {
-    return baseUrl
-  }
-  return `${baseUrl}/`
+function unslash (url) {
+  return url.replace(/([^:]\/)\/+/g, '$1')
 }
```

Comment:
I'm not sure what this is doing?

```node
> function unslash (url) {
...    return url.replace(/([^:]\/)\/+/g, '$1')
... }
undefined
> unslash('http://abc.def/g/h/ijk/l')
'http://abc.def/g/h/ijk/l'
```

Reply:
Ah, sorry, missing docs. Will add some tomorrow, but happy to rename it too. Wrote the function name without thinking too much about it, sorry.

```js
unslash('http://abc.def///g////h/ijk/l')
"http://abc.def/g/h/ijk/l"
```

Reply:
Ah, thanks, that makes sense, could call it `noramlise` or something like that.